pkg/policy/repository.go

```diff
@@ -369,3 +373,85 @@ func (p *Repository) NumRules() int {
 func (p *Repository) GetRevision() uint64 {
 	return p.revision
 }
+
+// ConvertToK8sServiceToToCIDR traverses all egress rules and matches them against provided serviceInfo. If a matching egress rule is found it is populated with ToCIDR and ToPorts entries based on endpoint object.
+func (p *Repository) ConvertToK8sServiceToToCIDR(serviceInfo types.K8sServiceNamespace, endpoint types.K8sServiceEndpoint) error {
```

Comment:
I would rather not polute `pkg/policy` with Kubernetes specific code. Instead, this should be generic functionality. In the future, this translation will happen for non Kubernetes use cases as well. What you want here is to introduce a concept of a `Translator`. It should be a go interface that `pkg/k8s` implemented. We then allow the translator to be called.

```
type Translator interface {
     TranslateRule(rule Rule)
}
```

```
func (p *Repository) RunTranslator(translator Translator) error {
    for _, rule := range p.rules {
        if err := translator.TranslateRule(rule); err != nil {
            return err
        }
    }
}
```

Reply:
Agreed, will do.

Reply:
Done